extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/QuarkusRestClientBuilder.java

```diff
@@ -0,0 +1,236 @@
+     *
+     * @param url the base Url for the service.
+     * @return the current builder with the baseUrl set.
+     */
+    QuarkusRestClientBuilder baseUrl(URL url);
+
+    /**
+     * Specifies the base URI to be used when making requests. Assuming that the interface has a
+     * <code>@Path("/api")</code> at the interface level and a <code>uri</code> is given with
+     * <code>http://my-service:8080/service</code> then all REST calls will be invoked with a <code>uri</code> of
+     * <code>http://my-service:8080/service/api</code> in addition to any <code>@Path</code> annotations included on the
+     * method.
+     *
+     * Subsequent calls to this method will replace the previously specified baseUri/baseUrl.
+     *
+     * @param uri the base URI for the service.
+     * @return the current builder with the baseUri set
+     * @throws IllegalArgumentException if the passed in URI is invalid
+     */
+    QuarkusRestClientBuilder baseUri(URI uri);
+
+    /**
+     * Set the connect timeout.
+     * <p>
+     * Like JAX-RS's <code>jakarta.ws.rs.client.ClientBuilder</code>'s <code>connectTimeout</code> method, specifying a
+     * timeout of 0 represents infinity, and negative values are not allowed.
+     * </p>
+     * <p>
+     * If the client instance is injected via CDI and the
+     * &quot;<em>fully.qualified.InterfaceName</em>/mp-rest/connectTimeout&quot; property is set via MicroProfile
+     * Config, that property's value will override, the value specified to this method.
+     * </p>
+     *
+     * @param timeout the maximum time to wait.
+     * @param unit the time unit of the timeout argument.
+     * @return the current builder with the connect timeout set.
+     * @throws IllegalArgumentException if the value of timeout is negative.
+     */
+    QuarkusRestClientBuilder connectTimeout(long timeout, TimeUnit unit);
+
+    /**
+     * Set the read timeout.
+     * <p>
+     * Like JAX-RS's <code>jakarta.ws.rs.client.ClientBuilder</code>'s <code>readTimeout</code> method, specifying a
+     * timeout of 0 represents infinity, and negative values are not allowed.
+     * </p>
+     * <p>
+     * Also like the JAX-RS Client API, if the read timeout is reached, the client interface method will throw a
+     * <code>jakarta.ws.rs.ProcessingException</code>.
+     * </p>
+     * <p>
+     * If the client instance is injected via CDI and the
+     * &quot;<em>fully.qualified.InterfaceName</em>/mp-rest/readTimeout&quot; property is set via MicroProfile Config,
+     * that property's value will override, the value specified to this method.
+     * </p>
+     *
+     * @param timeout the maximum time to wait.
+     * @param unit the time unit of the timeout argument.
+     * @return the current builder with the connect timeout set.
+     * @throws IllegalArgumentException if the value of timeout is negative.
+     */
+    QuarkusRestClientBuilder readTimeout(long timeout, TimeUnit unit);
```

Comment:
We should probably also have tests where these properties override the configured values in `application.properties`

Reply:
Removed

Reply:
This is covered by RestClientCDIDelegateBuilderTest